cpp/src/arrow/compute/kernels/scalar_temporal.cc

```diff
@@ -321,6 +323,68 @@ struct Nanosecond {
   }
 };
 
+// ----------------------------------------------------------------------
+// Convert timestamps to a string representation with an arbitrary format
+
+template <typename Duration>
+inline std::string get_timestamp(int64_t arg, const time_zone* tz,
+                                 const std::string* format) {
+  auto zt =
+      arrow_vendored::date::zoned_time<Duration>{tz, sys_time<Duration>(Duration{arg})};
+  return arrow_vendored::date::format(*format, zt);
+}
+
+template <typename Duration>
+struct Strftime {
+  static Status Call(KernelContext* ctx, const Scalar& in, Scalar* out) {
+    const auto& timezone = GetInputTimezone(in);
+    if (timezone.empty()) {
+      return Status::Invalid("Timezone naive timestamp can not be reliably printed.");
```

Comment:
```suggestion
      return Status::Invalid("Timestamps without a time zone cannot be reliably printed.");
```
Nit: We don't formally use "naive" in the docs that I'm aware of as it is a python-only concept.

Reply:
Agreed.